File: 프로그래머스/0/120923. 연속된 수의 합/연속된 수의 합.java
import java.util.*;
class Solution {
    public List<Integer> solution(int num, int total) {
        List<Integer> l = new ArrayList<>();
        for(int i = -1000; i - num <= 1000; i++){
            int sum = 0;
            int n = num;
            int j = i;
            l.clear();
            while(n-- > 0) {
                l.add(j); sum += j++;
            }
            if(sum == total) break;
        }
        return l;
    }
}

File: 프로그래머스/0/120923. 연속된 수의 합/연속된 수의 합.py
def solution(num, total):
    if num % 2 == 0:
        start = total // num - (num //2 -1)
    else:
        start = (total // num) - (num //2)
    return [x for x in range(start, start + num)]
    

"""
1부터해서
나누면 중앙이네 
짝수면 해당 숫자를 기준으로 앞 숫자부터 시작


"""